modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java

```diff
@@ -0,0 +1,269 @@
+                + ". response " + "is null.");
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Login response header is null for "
+                + application.getApplicationName());
+    }
+
+    private void testSendAuthenticationRequest(OIDCApplication application, HttpClient client) throws Exception {
+
+        List<NameValuePair> urlParameters = OIDCUtilTest.getNameValuePairs(application);
+        application.setApplicationContext(StringUtils.EMPTY);
+
+        HttpResponse response = sendPostRequestWithParameters(client, urlParameters, String.format(targetApplicationUrl,
+                application.getApplicationContext() + appUserAuthorizePath));
+        Assert.assertNotNull(response, "Authorization request failed for " + application.getApplicationName()
+                + ". " + "Authorized response is null");
+
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Authorization request failed for "
+                + application.getApplicationName() + ". Authorized response header is null");
+
+        EntityUtils.consume(response.getEntity());
+
+        response = sendGetRequest(client, locationHeader.getValue());
+        Assert.assertNotNull(response, "Authorization request failed for " + application.getApplicationName()
+                + ". " + "Authorized user response is null.");
+
+        Map<String, Integer> keyPositionMap = new HashMap<>(1);
+        keyPositionMap.put("name=\"sessionDataKey\"", 1);
+        List<DataExtractUtil.KeyValue> keyValues = DataExtractUtil.extractDataFromResponse(response, keyPositionMap);
+        Assert.assertNotNull(keyValues, "sessionDataKey key value is null for "
+                + application.getApplicationName());
+
+        sessionDataKey = keyValues.get(0).getValue();
+        Assert.assertNotNull(sessionDataKey, "Invalid sessionDataKey for " + application.getApplicationName());
+
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private void initAndCreatePlaygroundApplication() throws Exception {
+
+        playgroundApp = new OIDCApplication(playgroundAppName,
+                playgroundAppContext,
+                playgroundAppCallBackUri);
+        playgroundApp.addRequiredClaim(OIDCUtilTest.emailClaimUri);
+        playgroundApp.addRequiredClaim(OIDCUtilTest.firstNameClaimUri);
+        playgroundApp.addRequiredClaim(OIDCUtilTest.lastNameClaimUri);
+        ServiceProvider serviceProvider = new ServiceProvider();
+        createApplication(serviceProvider, playgroundApp);
+    }
+
+    private void addUserIntoJDBCUserStore(String username, String password, boolean isSecondaryStoreUser) throws Exception {
+
+        if (isSecondaryStoreUser) {
+            userMgtClient.addRole(secondaryUserRole, null, new String[]{PERMISSION_LOGIN});
```

Comment:
Why do you need roles, if the intension of the test is to check the application login?